aws_xray_sdk/core/models/entity.py

```diff
@@ -81,6 +81,7 @@ def add_subsegment(self, subsegment):
         """
         self._check_ended()
         subsegment.parent_id = self.id
+        subsegment.sampled = self.sampled if subsegment.sampled == True else False
```

Comment:
Why are we doing parent based sampling if the `subsegment` itself is sampled? Do we need this line at all?

Reply:
This is specifically for the case when a subsegment is added to an unsampled subsegment. We have to ensure that once an unsampled subsegment is created, all of its child subsegments are also unsampled. 

So for example, it is possible to do this:
```
    segment = Segment('parent_segment', 'id', 'id', True)
    subsegment = Subsegment('subsegment1', 'local', segment)
    subsegment.sampled = False
    segment.add_subsegment(subsegment)
    subsegment2 = Subsegment('subsegment2', 'local', segment)
    subsegment.add_subsegment(subsegment2)
```
In this case, it would be illegal for `subsegment2` to be sampled, so it has to take the decision of it's direct parent subsegment - `subsegment1`- which is unsampled.

Reply:
Does it make more sense to move this assignment to the "begin_subsegment" function in recorder.py and just base it off of the parent there? Then we can just base it off of the entity here "current_entity = self.get_trace_entity()". That entity would be the direct parent no?

Reply:
Having to check two levels like that is a code smell to me.

Reply:
There are two different ways to create subsegments, which is why this had to be done on two different levels - @srprash please correct me if I'm missing something